core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -46,37 +48,14 @@ object WorkflowWebsocketResource {
   val sessionJobs = new mutable.HashMap[String, (WorkflowCompiler, ActorRef)]
 
   // Map[sessionId, Map[operatorId, List[ITuple]]]
-  val sessionResults = new mutable.HashMap[String, Map[String, List[ITuple]]]
+  val sessionResults = new mutable.HashMap[String, WorkflowResultService]
 
   // Map[sessionId, Map[downloadType, googleSheetLink]
   val sessionDownloadCache = new mutable.HashMap[String, mutable.HashMap[String, String]]
 
-  /**
-    * Calculate which page in frontend need to be re-fetched
-    * @param beforeList data before status update event (i.e. unmodified sessionResults)
-    * @param afterList data after status update event
-    * @return list of indices of modified pages starting from 1
-    */
-  def getDirtyPageIndices(beforeList: List[ITuple], afterList: List[ITuple]): List[Int] = {
-    val pageSize = 10
-
-    var currentIndex = 1
-    var currentIndexPageCount = 0
-    val dirtyPageIndices = new mutable.HashSet[Int]()
-    for ((before, after) <- beforeList.zipAll(afterList, null, null)) {
-      if (before == null || after == null || !before.equals(after)) {
-        dirtyPageIndices.add(currentIndex)
-      }
-      currentIndexPageCount += 1
-      if (currentIndexPageCount == pageSize) {
-        currentIndexPageCount = 0
-        currentIndex += 1
-      }
-    }
-
-    dirtyPageIndices.toList
+  def send(session: Session, event: TexeraWebSocketEvent): Unit = {
```

Comment:
I see this is just moved here.

this `send` seems to be a very general helper method to send whatever `TexeraWebSocketEvent` through websocket. Can we document its usage (add some comments) to make sure the users of this method can be explicit?